fieldpath/element.go

```diff
@@ -123,17 +172,16 @@ func (e PathElement) String() string {
 // names (type must be string) with values (type must be value.Value). If these
 // conditions are not met, KeyByFields will panic--it's intended for static
 // construction and shouldn't have user-produced values passed to it.
-func KeyByFields(nameValues ...interface{}) []value.Field {
+func KeyByFields(nameValues ...interface{}) []KeyValue {
 	if len(nameValues)%2 != 0 {
 		panic("must have a value for every name")
 	}
-	out := []value.Field{}
+	out := []KeyValue{}
 	for i := 0; i < len(nameValues)-1; i += 2 {
-		out = append(out, value.Field{
-			Name:  nameValues[i].(string),
-			Value: nameValues[i+1].(value.Value),
-		})
+		out = append(out, KeyValue{Key: nameValues[i].(string), Value: nameValues[i+1]})
 	}
+	SortKeyValues(out)
```

Comment:
I don't claim it's better without looking at all call sites, but it's conceivable that we could panic here if they provided entries in the wrong order, since it may be misleading (for the humans reading failing tests) to specify things in one order and have them come out in a different order.